src/openvic-simulation/history/Bookmark.cpp

```diff
@@ -0,0 +1,95 @@
+#include "Bookmark.hpp"
+
+#include <openvic-dataloader/v2script/AbstractSyntaxTree.hpp>
+
+#include "openvic-simulation/dataloader/NodeTools.hpp"
+
+#include "types/Date.hpp"
+#include "types/IdentifierRegistry.hpp"
+#include "utility/Logger.hpp"
+
+using namespace OpenVic;
+using namespace OpenVic::NodeTools;
+
+Bookmark::Bookmark(
+	std::string_view new_identifier,
+	std::string_view new_name,
+	std::string_view new_description,
+	Date new_date,
+	uint32_t new_initial_camera_x,
+	uint32_t new_initial_camera_y
+)
+	: HasIdentifier { new_identifier },
+	  name { new_name },
+	  description { new_description },
+	  date { new_date },
+	  initial_camera_x { new_initial_camera_x },
+	  initial_camera_y { new_initial_camera_y } {
+}
+
+std::string_view Bookmark::get_name() const {
+	return name;
+}
+
+std::string_view Bookmark::get_description() const {
+	return description;
+}
```

Comment:
Where possible, I've been returning `std::string const&` rather than `std::string_view`s - if a `std::string_view` is needed then it'll be converted automatically, we don't need to force that conversion now, especially if we ever need these values as proper `std::string`s or C strings.

Reply:
A const string reference can potentially allocate still, whereas string_views are guaranteed never to allocate, in every case where you know the string will outlive a call, you should be using string_view as it will never allocate.